src/commands/command_handlers.py

```diff
@@ -26,3 +27,18 @@ async def snap(ctx: Context, channel: discord.TextChannel = None, length: int =
         @self.bot.command(name='bing')
         async def snap(ctx: Context):
             await ctx.channel.send("bong")
+
+        @self.bot.command(name='cache')
+        async def cache(ctx: Context, key, val):
+            self.cache[key] = val
+
+        @self.bot.command(name='showcache')
+        async def showcache(ctx: Context):
+            await ctx.channel.send(self.cache)
+
+        @self.bot.command(name='mirror')
```

Comment:
I think you can probably collapse the other helper functions here with some mildly clever *args parsing and breaking it out into its own cog 

https://discordpy.readthedocs.io/en/stable/ext/commands/commands.html#variable
https://discordpy.readthedocs.io/en/stable/ext/commands/cogs.html

Reply:
moved to a cog + collapsed cache into a single cache function. it's going to just be here for testing for the time being